all-sdk/autotuning.md

```diff
@@ -0,0 +1,431 @@
+# Worker Autotuning Proposal
+
+Users often struggle with the appropriate configuration of workers. There are many options to fiddle with, nearly none
+of which are something users should be required to think about. This proposal aims to reduce the cognitive load of
+configuring workers by automatically tuning various worker internals at runtime without (or with configurable) user
+intervention.
+
+## Areas of configuration
+There are a few different configuration areas that we can autotune:
+
+### Concurrent Pollers
+Right now the number of pollers used for polling workflow and activity tasks is fixed to a maximum defined by the user.
+In Core: `max_concurrent_wft_polls` and `max_concurrent_at_polls`. There's no need for this number to be fixed, and we
+can and should scale it up and down as demand dictates.  This work is largely already designed and implemented in Go:
+[see the branch
+here](https://github.com/temporalio/sdk-go/compare/master...Quinn-With-Two-Ns:sdk-go:poller-autottune-v0.2). Hence this
+doc will not focus on this area in detail.
+
+It is important to note that there is a distinction between the active number of poller *routines* and the active number
+of *open actual RPC calls*. The number of routines may be scaled independently of the number of open RPC calls at any
+time. The former may have a floor, whereas the latter must always be able to go to 0. Users are unlikely to care about
+the number of routines as they are relatively cheap. They *do*, however, care about how quickly we will change the
+number of active RPCs. The floor on the number of routines effectively determines how quickly we are willing to consume
+available slots and thus make actual RPC calls. IE: If the floor on the number of routines is 10, and 10 slots open up
```

Comment:
This is the first time the term "slots" is used -- a definition would be helpful.